openrtx/src/rtx/rtx.cpp

```diff
@@ -126,6 +128,74 @@ void rtx_task()
 
         pthread_mutex_unlock(cfgMutex);
     }
+     /* Spectrum update block, run when in SPECTRUM mode.
+*
+* The spectrum mode is a special mode where the radio is in RX mode but
+* the audio path is disabled. This allows to display the RSSI level of the
+* received signals across a frequency range in a waterfall-like display.
+* 
+* This block writes the received RSSI levels to the spectrum buffer.
+*/
+    #ifdef CONFIG_SPECTRUM
+    if(state.rtxStatus == RTX_SPECTRUM)
+    {
+        //state.spectrum_peakRssi = -160;
+        #define SPECTRUM_WF_LINES 64 // must match NUMBER_BARS in ui_menu.c!
+        uint32_t spectrumStep = freq_steps[state.settings.spectrum_step];
+        // Get the current RSSI level
+
+        uint8_t peakIndex;
+        // Write the RSSI level to the spectrum buffer
+        for (int i = 0; i < SPECTRUM_WF_LINES; i++)
+        {
+            // set freq to (state.spectrum_startFreq + i * spectrumStep), no bk4819
+            rtxStatus.rxFrequency = (state.spectrum_startFreq + i * spectrumStep);
+            rtxStatus.txFrequency = rtxStatus.rxFrequency;
+            rssi = radio_getRssi();
+            uint8_t height = (rssi + 160) / 2;
+            // Macro for log2, not using the math library
+            #define log2(x) (31 - __builtin_clz(x))
+            //uint8_t height = ((rssi + 160)*log2(22 - (rssi>>1) )) >> 3;
+            state.spectrum_data[i] = height;
+            // set peak value
+            if(rssi > state.spectrum_peakRssi)
+            {
+                state.spectrum_peakRssi = rssi;
+                state.spectrum_peakFreq = state.spectrum_startFreq + i * spectrumStep;
+                state.spectrum_peakIndex = i;
+            }
+            // stop scanning if the rssi is greater than the current squelch rssi,
+            // and listen to that frequency
+            if(radio_getRssi() > (-127 + (state.settings.sqlLevel * 66) / 15))
```

Comment:
Do we really want this? By the way, the squelch is already managed in OpMode_FM and this is conflicting with that part

Reply:
Noted, will remove.